src/github.com/getlantern/peerscanner/cfr/cfr_test.go

```diff
@@ -0,0 +1,61 @@
+package cfr
+
+import (
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/satori/go.uuid"
+
+	"github.com/getlantern/fdcount"
+	"github.com/getlantern/testify/assert"
+)
+
+func TestList(t *testing.T) {
+	_, counter, err := fdcount.Matching("TCP")
+	if err != nil {
+		t.Fatalf("Unable to get starting fdcount: %v", err)
+	}
+	cfr := New(os.Getenv("CFR_ID"), os.Getenv("CFR_KEY"), httpClientWithDisabledKeepAlives())
+	dists, err := ListDistributions(cfr)
+	if assert.NoError(t, err, "Should be able to get all distributions") {
+		for _, d := range dists {
+			log.Tracef("%v : %v (%v)", d.InstanceId, d.Domain, d.Status)
+		}
+		assert.True(t, len(dists) > 0, "There should be some distributions")
+	}
+	assert.NoError(t, counter.AssertDelta(0), "All file descriptors should have been closed")
+}
+
+func TestCreateAndRefresh(t *testing.T) {
+	_, counter, err := fdcount.Matching("TCP")
+	if err != nil {
+		t.Fatalf("Unable to get starting fdcount: %v", err)
+	}
+	cfr := New(os.Getenv("CFR_ID"), os.Getenv("CFR_KEY"), httpClientWithDisabledKeepAlives())
+	// Deleting cloudfront distributions is actually quite an involved process.
+	// Fortunately, distributions per se cost us nothing.  A separate service
+	// will be implemented to delete test and otherwise unused distributions.
+	name := uuid.NewV4().String()
+	dist, err := CreateDistribution(cfr, name, name+"-grey.flashlightproxy.org", "TEST -- DELETE")
+	assert.NoError(t, err, "Should be able to create distribution")
+	assert.Equal(t, "InProgress", dist.Status, "New distribution should have Status: \"InProgress\"")
+	assert.Equal(t, name, dist.InstanceId, "New distribution should have the right InstanceId")
+	assert.True(t, strings.HasSuffix(dist.Domain, ".cloudfront.net"), "Domain should be a .cloudfront.net subdomain, not '"+dist.Domain+"'")
+	err = RefreshStatus(cfr, dist)
+	assert.NoError(t, err, "Should be able to refresh status")
+	// Just check that Status stays a valid one.  Checking that it eventually
+	// gets refreshed to "Deployed" would take a few minutes, and thus is out
+	// of the scope of this unit test.
+	assert.Equal(t, "InProgress", dist.Status, "New distribution should have Status: \"InProgress\" even after refreshing right away")
+	assert.NoError(t, counter.AssertDelta(0), "All file descriptors should have been closed")
+}
+
```

Comment:
I ran ``go test`` without realizing I didn't have CFR_KEY or CFR_ID set. It might be nice to display a warning instead of proceeding ahead and trying to deference a nil pointer? At least a more suitable error message would be helpful in this scenario.
```
--- FAIL: TestCreateAndRefresh (0.13s)
        Location:       cfr_test.go:42
	Error:		No error is expected but got The security token included in the request is invalid.
	Messages:	Should be able to create distribution

panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x6719c]

goroutine 12 [running]:
testing.func·006()
	/usr/local/go/src/testing/testing.go:441 +0x181
github.com/getlantern/peerscanner/cfr.TestCreateAndRefresh(0xc208296ea0)
	/Users/todd/Dev/go/src/github.com/getlantern/flashlight-build/src/github.com/getlantern/peerscanner/cfr/cfr_test.go:43 +0x11ac
testing.tRunner(0xc208296ea0, 0x57b458)
	/usr/local/go/src/testing/testing.go:447 +0xbf
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:555 +0xa8b
```